Wyam.Core/Modules/FileName.cs

```diff
@@ -83,8 +85,13 @@ public FileName(string inputKey, string outputKey)
             _fileName = (d, c) => d.String(inputKey);
             _outputKey = outputKey;
         }
-        
-        public FileName(DocumentConfig fileName, string outputKey)
+
+		public FileName(IEnumerable<string> allowedCharacters) : this()
```

Comment:
Good idea to add a whitelist - can we make this a fluent method instead: `WithAllowedChars()`? That way you can specify allowed characters regardless of which keys you specify to get or set in the existing constructors.